File: src/game/level/world1/World1Level1.java
package game.level.world1;

import game.entity.enemies.boss.Boss;
import game.entity.enemies.npcs.FlyingEnemy;
import game.entity.enemies.npcs.YellowBall;
import game.entity.models.Enemy;
import game.level.Level;
import game.level.world2.World2Level;

/**

 Represents the first level of the game, extending the abstract class Level.

 Provides specific implementations of methods for level 1, including images for the blocks,

 explosion length, and a method to generate the stone blocks in the game board.
 */
public class World1Level1 extends World1Level {
    public static final int LEVEL_ID = 1;

    @Override
    public int getLevelId() {
        return LEVEL_ID;
    }

    @Override
    public int startEnemiesCount() {
        return 5;
    }

    @Override
    public Class<? extends Enemy>[] availableEnemies() {
        return new Class[]{
                YellowBall.class
        };
    }

    @Override
    public Class<? extends Level> getNextLevel() {
        return World1Level2.class;
    }
}

File: src/game/level/world1/World1Level1.kt
package game.level.world1

import game.entity.enemies.npcs.YellowBall
import game.entity.models.Enemy
import game.level.Level
import game.level.StoryLevel
import game.level.info.model.LevelInfo
import game.level.info.imp.World1LevelInfo

class World1Level1 : StoryLevel() {
    override val info: LevelInfo
        get() = object : World1LevelInfo(this) {
            override val levelId: Int get() = 1
            override val startEnemiesCount: Int get() = 7
            override val availableEnemies: Array<Class<out Enemy>> get() = arrayOf(YellowBall::class.java)
            override val nextLevel: Class<out Level?> get() = World1Level2::class.java
        }
}